### PR TITLE
Address code smells in DataGenerator

### PR DESCRIPTION
**Pull request by peternied**
### Description
Address code smells in DataGenerator
- Defined constants for reused literals
- Mark unmodified fields final
- Add a private constructor to hide the implicit public constructor

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gregschohn — COMMENTED** [review]
UtilityClass is one annotation I'd rather us not use.  It's value is marginal and it requires me to review the entire class/file, which is especially scary when looking at a diff since a reviewer won't see the whole file by default.

**Comment by peternied** [comment]
> UtilityClass is one annotation I'd rather us not use. It's value is marginal and it requires me to review the entire class/file, which is especially scary when looking at a diff since a reviewer won't see the whole file by default.

Thanks for the review, I have addressed this comment.

File: DataGenerator/src/main/java/org/opensearch/migrations/data/workloads/Geonames.java [hunk 1]
```diff
@@ -9,6 +9,9 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
+import static org.opensearch.migrations.data.FieldBuilders.GEO_POINT;
+import static org.opensearch.migrations.data.FieldBuilders.INTEGER;
+import static org.opensearch.migrations.data.FieldBuilders.LONG;
```
Comment (gregschohn) [inline DataGenerator/src/main/java/org/opensearch/migrations/data/workloads/Geonames.java:14]: Let's continue to not allow static imports outside of test code (this is in src, not testFixture)
Adding the classname will give names like "INTEGER" some more context and clarity

Comment (peternied) [reply DataGenerator/src/main/java/org/opensearch/migrations/data/workloads/Geonames.java:14]: Thanks for the feedback, I don't agree with the style preference.  I would prefer to merge this change as is. 

I've created MIGRATIONS-2141 [1] to automate enforcement of style issues, lets use tooling to align our preferences.

- [1] https://opensearch.atlassian.net/browse/MIGRATIONS-2141

**Review by lewijacn — APPROVED** [review]

**Pull request merged**